x-pack/plugins/security_solution/server/lists_integration/endpoint/handlers/exceptions_pre_delete_item_handler.ts

```diff
@@ -5,14 +5,45 @@
  * 2.0.
  */
 
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
 import { EndpointAppContextService } from '../../../endpoint/endpoint_app_context_services';
-import { ExtensionPoint } from '../../../../../lists/server';
+import { ExceptionsListPreDeleteItemServerExtension } from '../../../../../lists/server';
+import { TrustedAppValidator } from '../validators/trusted_app_validator';
 
+type ValidatedReturnType = ExceptionsListPreDeleteItemServerExtension['callback'];
 export const getExceptionsPreDeleteItemHandler = (
-  endpointAppContext: EndpointAppContextService
-): (ExtensionPoint & { type: 'exceptionsListPreDeleteItem' })['callback'] => {
-  return async function ({ data }) {
-    // Individual validators here
+  endpointAppContextService: EndpointAppContextService
+): ValidatedReturnType => {
+  return async function ({ data, context: { request, exceptionListClient } }) {
+    if (data.namespaceType !== 'agnostic') {
+      return data;
+    }
+
+    const exceptionItem: ExceptionListItemSchema | null =
+      await exceptionListClient.getExceptionListItem({
+        id: data.id,
+        itemId: data.itemId,
+        namespaceType: data.namespaceType,
+      });
+
+    if (!exceptionItem) {
+      return data;
+    }
+
+    // Validate Trusted Applications
+    if (TrustedAppValidator.isTrustedApp({ listId: exceptionItem.list_id })) {
+      const trustedAppValidatorInstance = new TrustedAppValidator(
+        endpointAppContextService,
+        request
+      );
+
+      if (data.itemId) {
+        await trustedAppValidatorInstance.validatePreDeleteItemById();
+      } else {
+        await trustedAppValidatorInstance.validatePreDeleteItem();
+      }
```

Comment:
@paul-tavares perhaps we don't need a `preDeleteItemById`? I went with the idea that deleteByItemId should need an `itemId` 😄

Reply:
thats correct - both of the `ExceptionListClient` methods will execute the `exceptionsListPreDeleteItem` callbacks, so you don't need to have a different `validate*` method for each one